backend/api/v1/rollout_service.go

```diff
@@ -0,0 +1,281 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/bytebase/bytebase/backend/common"
+	"github.com/bytebase/bytebase/backend/store"
+	storepb "github.com/bytebase/bytebase/proto/generated-go/store"
+	v1pb "github.com/bytebase/bytebase/proto/generated-go/v1"
+)
+
+// RolloutService represents a service for managing rollout.
+type RolloutService struct {
+	v1pb.UnimplementedRolloutServiceServer
+	store *store.Store
+}
+
+// NewRolloutService returns a rollout service instance.
+func NewRolloutService(store *store.Store) *RolloutService {
+	return &RolloutService{
+		store: store,
+	}
+}
+
+// GetPlan gets a plan.
+func (s *RolloutService) GetPlan(ctx context.Context, request *v1pb.GetPlanRequest) (*v1pb.Plan, error) {
+	planID, err := getPlanID(request.Name)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+	plan, err := s.store.GetPlan(ctx, planID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get plan, error: %v", err)
+	}
+	if plan == nil {
+		return nil, status.Errorf(codes.NotFound, "plan not found for id: %d", planID)
+	}
+	return convertToPlan(plan), nil
+}
+
+// CreatePlan creates a new plan.
+func (s *RolloutService) CreatePlan(ctx context.Context, request *v1pb.CreatePlanRequest) (*v1pb.Plan, error) {
+	principalUID := ctx.Value(common.PrincipalIDContextKey).(int)
+	projectID, err := getProjectID(request.Parent)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+	planMessage := &store.PlanMessage{
+		ProjectID:   projectID,
+		PipelineUID: nil,
+		Name:        request.Plan.Title,
+		Description: request.Plan.Description,
+		Config: &storepb.PlanConfig{
+			Steps: convertPlanSteps(request.Plan.Steps),
+		},
+	}
+	plan, err := s.store.CreatePlan(ctx, planMessage, principalUID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to create plan, error: %v", err)
+	}
+	return convertToPlan(plan), nil
+}
+func convertToPlan(plan *store.PlanMessage) *v1pb.Plan {
+	return &v1pb.Plan{
+		Name:        fmt.Sprintf("%s%s/%s%d", projectNamePrefix, plan.ProjectID, planPrefix, plan.UID),
+		Uid:         fmt.Sprintf("%d", plan.UID),
```

Comment:
Why do we need an extra field `Uid`, we can get the `plan.UID` in `Name`?

Reply:
https://google.aip.dev/148#uid

not a UUIDv4 though...